src/opengeodeweb_back/geode_functions.py

```diff
@@ -16,34 +16,131 @@
+        data, builder, name, input_coordiante_system, output_coordiante_system
+    )
+
+
+def is_model(geode_object: str):
+    return objects_list()[geode_object]["is_model"]
+
+
+def is_3D(geode_object: str):
+    return objects_list()[geode_object]["is_3D"]
+
+
+def is_viewable(geode_object: str):
+    return objects_list()[geode_object]["is_viewable"]
+
+
+def get_inspector(geode_object: str, data):
+    return objects_list()[geode_object]["inspector"](data)
+
+
+def save_viewable(geode_object: str, data, folder_absolute_path: str, id: str):
+    return objects_list()[geode_object]["save_viewable"](
+        data, os.path.join(folder_absolute_path, id)
+    )
+
+
+def get_geode_object_input_extensions(geode_object: str):
+    inputs_list = []
+    geode_object_inputs = get_input(geode_object)
+
+    for input in geode_object_inputs:
+        list_creators = input.list_creators()
+        inputs_list = inputs_list + list_creators
+    inputs_list = list(set(inputs_list))
+    inputs_list.sort()
+    return inputs_list
+
+
+def get_geode_object_output_extensions(geode_object: str):
+    output_list = []
+    geode_object_outputs = get_output(geode_object)
+
+    for output in geode_object_outputs:
+        list_creators = output.list_creators()
+        output_list = output_list + list_creators
+    output_list = list(set(output_list))
+    output_list.sort()
+    return output_list
+
+
+def list_input_extensions(
+    keys: list = [],
+):
     """
     Purpose:
         Function that returns a list of all input extensions
     Args:
-        crs -- Tells the function if we want the geode_objects that have a crs
+        keys -- Tells the function if we want the geode_objects that have a crs
+        geode_object -- The name of the geode_object
     Returns:
         An ordered list of input file extensions
     """
-    List = []
-    geode_object_dict = objects_list()
+    extensions_list = []
 
-    for geode_object in geode_object_dict.values():
-        values = geode_object["input"]
+    for geode_object, value in objects_list().items():
+        if keys:
+            for key in keys:
+                if key in geode_object:
+                    if type(geode_object[key]) == bool and geode_object[key] == True:
+                        continue
+                    else:
+                        continue
+                else:
+                    continue
 
-        # if crs == True:
-        #     if "crs" not in geode_object:
-        #         continue
-        for value in values:
-            list_creators = value.list_creators()
-            for creator in list_creators:
-                if creator not in List:
-                    List.append(creator)
-    List.sort()
-    return List
+        geode_object_list_inputs = get_geode_object_input_extensions(geode_object)
+        extensions_list = extensions_list + geode_object_list_inputs
 
+    extensions_list = list(set(extensions_list))
+    extensions_list.sort()
```

Comment:
Why not create a set in the first place?
and then `return sorted(extensions_list)`